client/app.jsx

```diff
@@ -1,9 +1,36 @@
 import React from 'react';
 import Home from './pages/home';
+import Results from './pages/results';
+import parseRoute from './lib/parse-route';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
 
 export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      route: parseRoute(window.location.hash)
+    };
+  }
+
+  componentDidUpdate() {
+    window.addEventListener('hashchange', () => {
+      const parsedRoute = parseRoute(window.location.hash);
+      this.setState({ route: parsedRoute });
+    });
+  }
```

Comment:
If you are using react router you can delete all of the hash routing related code.